modules/ingest-common/src/main/java/org/opensearch/ingest/common/IngestCommonModulePlugin.java

```diff
@@ -147,4 +157,27 @@ private static MatcherWatchdog createGrokThreadWatchdog(Processor.Parameters par
         );
     }
 
+    private Map<String, Processor.Factory> filterForAllowlistSetting(Settings settings, Map<String, Processor.Factory> map) {
+        if (PROCESSORS_ALLOWLIST_SETTING.exists(settings) == false) {
+            return Map.copyOf(map);
+        }
+        final Set<String> allowlist = Set.copyOf(PROCESSORS_ALLOWLIST_SETTING.get(settings));
+        // Assert that no unknown processors are defined in the allowlist
+        final Set<String> unknownAllowlistProcessors = allowlist.stream()
+            .filter(p -> map.containsKey(p) == false)
+            .collect(Collectors.toSet());
+        if (unknownAllowlistProcessors.isEmpty() == false) {
```

Comment:
```suggestion
        if (!unknownAllowlistProcessors.isEmpty()) {
```

Reply:
While I would love to agree with you, we have actually codified the opposite: https://github.com/opensearch-project/OpenSearch/blob/2a55a2a0c61e264d4d68759af9f82ea5e5e0e7d5/DEVELOPER_GUIDE.md?plain=1#L353